src/app/components/canvas/canvas.component.ts

```diff
@@ -34,34 +47,39 @@ export class CanvasComponent implements OnInit, OnDestroy {
     let shapes = { rect: 'Rectangle', triangle: 'Triangle', circle: 'Circle' };
     this.canvas.on('object:added', (options) => {
       if (options.target) {
-        this.eventService.sendEvent(
-          'Added ' + shapes[options.target.type as keyof typeof shapes]
-        );
+        var eventStr =
+          'Added ' + shapes[options.target.type as keyof typeof shapes];
+        this.eventService.sendEvent(eventStr);
+        //this.store.dispatch(onAdding());
+        this.updateCanvasState(eventStr);
       }
     });
     this.canvas.on('object:moving', (options) => {
       if (options.target) {
-        this.eventService.sendEvent(
-          'Translated ' + shapes[options.target.type as keyof typeof shapes]
-        );
+        var eventStr =
+          'Translated ' + shapes[options.target.type as keyof typeof shapes];
+        this.eventService.sendEvent(eventStr);
+        this.updateCanvasState(eventStr);
       }
     });
     this.canvas.on('object:rotating', (options) => {
       if (options.target) {
-        this.eventService.sendEvent(
-          'Rotated ' + shapes[options.target.type as keyof typeof shapes]
-        );
+        var eventStr =
+          'Rotated ' + shapes[options.target.type as keyof typeof shapes];
+        this.eventService.sendEvent(eventStr);
+        this.updateCanvasState(eventStr);
       }
     });
     this.canvas.on('object:scaling', (options) => {
       if (options.target) {
-        this.eventService.sendEvent(
-          'Scaled ' + shapes[options.target.type as keyof typeof shapes]
-        );
+        var eventStr =
+          'Scaled ' + shapes[options.target.type as keyof typeof shapes];
+        this.eventService.sendEvent(eventStr);
+        this.updateCanvasState(eventStr);
       }
     });
   }
```

Comment:
Can you try to make you of 'modified' event rather than using rotating, scaling and moving events to update the store, as the aforementioned event keeps firing at every instant.